src/core/ipns/resolver.js

```diff
@@ -96,14 +98,15 @@ class IpnsResolver {
       return callback(err)
     }
 
-    const { routingKey } = ipns.getIdKeys(peerId.toBytes())
+    const { routingKey, routingPubKey } = ipns.getIdKeys(peerId.toBytes())
 
-    // TODO DHT - get public key from routing?
-    // https://github.com/ipfs/go-ipfs/blob/master/namesys/routing.go#L70
-    // https://github.com/libp2p/go-libp2p-routing/blob/master/routing.go#L99
-
-    this._routing.get(routingKey.toBuffer(), (err, res) => {
-      if (err) {
+    auto({
+      // Name should be the hash of a public key retrievable from ipfs.
+      // We retrieve public key to add it to the PeerId, as the IPNS record may not have it.
+      pubKey: (cb) => this._routing.get(routingPubKey.toBuffer(), cb),
+      record: (cb) => this._routing.get(routingKey.toBuffer(), cb)
```

Comment:
These will be done in parallel - is that the desired behaviour? If so you can probably just use `async/parallel` instead.